v-next/hardhat/src/internal/builtin-plugins/solidity-test/hook-handlers/test.ts

```diff
@@ -0,0 +1,22 @@
+import type { TestHooks } from "hardhat/types/hooks";
+
+import { resolveFromRoot } from "@nomicfoundation/hardhat-utils/path";
+
+export default async (): Promise<Partial<TestHooks>> => {
+  const handlers: Partial<TestHooks> = {
+    registerFileForTestRunner: async (context, filePath, next) => {
+      const absoluteFilePath = resolveFromRoot(process.cwd(), filePath);
+
+      if (
+        absoluteFilePath.includes(context.config.paths.tests.solidity) &&
```

Comment:
```suggestion
        absoluteFilePath.includes(context.config.paths.tests.solidity) ||
```

Reply:
maybe a nitpick, you could use `startsWith` instead of `includes` in all the checks.